Initial thoughts on features Transpower might like in OpenSCD

danyill opened the pull request:
Some initial thoughts.

danyill committed:
Initial thoughts

danyill committed:
Add requirement for IID export

danyill committed:
Noticed we needed the NeutralPoint

danyill committed:
Add additional extension

danyill committed:
Update ROADMAP.md

Add requirements for subscriber view of later binding plugins

jarradraumati commented on ROADMAP.md:
```diff
@@ -25,3 +25,62 @@ Feeature ideas for the OpenSCD project. They are currently not prioritized.
+   - Having a plugin to export relevant information to DCIM/IPAM tools (mainly IP addresses) so configuration can be built by importing it
+   - Having a plugin to export relevant information to Ansible network configuration scripting (to provide the information on what each device needs for VLANs, MAC address filtering, an augmented version of the Communication section)
+   - Automation of configuration for GOOSE and SV traffic based on known configurations of merging units, protection relays. We anticipate doing this by transforming the SLD into a graph where the nodes represent terminals and the edges represent `ConductingEquipment` etc and then carrying out a terminating depth-first search to identify e.g. CTs and CBs on a bus and provide subscriptions between ...
+     - to be able to instantiate project icd files against LNodes within the substation section.
```
```suggestion
     - to be able to bind instantiated IEDs `LN`s against `LNode`s within the substation/process section.
```

jarradraumati commented on ROADMAP.md:
```diff
@@ -25,3 +25,62 @@ Feeature ideas for the OpenSCD project. They are currently not prioritized.
+
+1. Substation Editor. The Substation editor provides much of the desired capability but does not yet allow:
+
+   - Connection of a an IED to a set of LNodes. (see [#1128](https://github.com/openscd/open-scd/issues/1128))
```
```suggestion
   - Connection of an IED LNs to LNodes with the substation/process section. (see [#1128](https://github.com/openscd/open-scd/issues/1128))
```

jarradraumati commented on ROADMAP.md:
```diff
@@ -25,3 +25,62 @@ Feeature ideas for the OpenSCD project. They are currently not prioritized.
+
+1. Complete the publisher plugin 
+
+   - Provide the ability to add FCDOs to RCBs to allow MMS SCADA map augmentation ([#1092](https://github.com/openscd/open-scd/issues/1092))
```
```suggestion
   - Provide the ability to add FCDAs to RCBs to allow MMS SCADA map augmentation ([#1092](https://github.com/openscd/open-scd/issues/1092))
```

danyill replied:
I think it's actually FCDOs. We can already do FCDAs in OpenSCD, but we can't drop entire data objects in like we do in other tools.

jarradraumati replied:
Ah alright. FCDO isn't mentioned within IEC 61850, so some may get confused. But it looks like other third parties have started using that term to define a functionally constrained data object.

This is the definition of FCDA:

> A reference to a single data attribute, a sub data object or sub data attribute of a data object having a specific functional constraint (FC) value shall be called functional constrained data attribute (FCDA).

jarradraumati reviewed (commented):
Suggested some amendments.

danyill committed:
Apply suggestions from code review

Thanks @jarradraumti for the feedback

Co-authored-by: Jarrad Raumati <jarrad.raumati@gmail.com>

danyill committed:
Update ROADMAP.md

Add feedback from David, add more detail on other Substation element requirements.

danyill committed:
Update ROADMAP.md

Tidying up and proofing

danyill committed:
Update ROADMAP.md

Make mention of how Transpower's business model may be differnent to European utilities

danyill committed:
Update ROADMAP.md

Final tweaks

danyill commented:
This has served it's purpose. Closing.
